apps/nextjs/src/utils/useClientSideFeatureFlag.ts

```diff
@@ -1,57 +1,17 @@
 import { useEffect, useState } from "react";
 
-import { useUser } from "@clerk/nextjs";
-
 import useAnalytics from "@/lib/analytics/useAnalytics";
 
-import { checkFeatureFlag } from "./checkFeatureFlag";
-
-let isUserIdentified = false;
-export function useClientSideFeatureFlag(
-  featureFlagId: string,
-): [boolean, boolean] {
-  const { user, isLoaded } = useUser();
+export function useClientSideFeatureFlag(flag: string): boolean {
+  const { posthogAiBetaClient: client } = useAnalytics();
 
-  const { posthogAiBetaClient } = useAnalytics();
-
-  const [isEnabled, setIsEnabled] = useState<boolean>(false);
-  const [isCheckComplete, setIsCheckComplete] = useState<boolean>(false);
+  const [featureEnabled, setFeatureEnabled] = useState<boolean | undefined>();
 
   useEffect(() => {
-    function checkFeatureFlagStatus() {
-      if (process.env.NEXT_PUBLIC_ENVIRONMENT !== "prd") {
-        setIsEnabled(true);
-        setIsCheckComplete(true);
-        return;
-      }
-
-      if (isLoaded && user) {
-        if (!isUserIdentified) {
-          const distinctId = posthogAiBetaClient.get_distinct_id();
-
-          if (distinctId !== user.id) {
-            posthogAiBetaClient.identify(user.id, {
-              email: user.primaryEmailAddress?.emailAddress,
-            });
-          }
-          isUserIdentified = true;
-        }
-        const isFeatureFlagEnabled = checkFeatureFlag(
-          posthogAiBetaClient,
-          featureFlagId,
-        );
-
-        setIsEnabled(isFeatureFlagEnabled);
-        setIsCheckComplete(true);
-      } else if (isLoaded && !user) {
-        console.log("User is not signed in");
-        setIsEnabled(false);
-        setIsCheckComplete(true);
-      }
-    }
-
-    checkFeatureFlagStatus();
-  }, [user, isLoaded, featureFlagId, posthogAiBetaClient]);
+    return client.onFeatureFlags(() => {
+      setFeatureEnabled(client.isFeatureEnabled(flag));
+    });
+  }, [client, flag]);
```

Comment:
This hook is a pretty close copy of the [official posthog hook ](https://github.com/PostHog/posthog-js/blob/26150b197dc1a2f5241b8d29ee7a359331a1a0a4/react/src/hooks/useFeatureFlagEnabled.ts)

Reply:
Hmm wonder why we aren't using there's

Reply:
Actually tbf there's it seems we don't control which posthog instance it uses

Reply:
If we use theirs it would mean adding their provider and providing `client` to it. But only one of the clients which could be confusing. The actual code in their library is pretty minimal though